Sources/SourceControl/GitRepository.swift

```diff
@@ -406,15 +406,17 @@ private class GitFileSystemView: FileSystem {
         }
     }
     
-    func getDirectoryContents(_ path: AbsolutePath) throws -> [String] {
+    func getDirectoryContents(_ path: AbsolutePath) throws -> AnySequence<String> {
         guard let entry = try getEntry(path) else {
             throw FileSystemError.noEntry
         }
         guard entry.type == .tree else {
             throw FileSystemError.notDirectory
         }
 
-        return try getTree(entry.hash).contents.map{ $0.name }
+        let contents = try getTree(entry.hash).contents.map{ $0.name }
+        // FIXME: Implement lazy behavior here.
```

Comment:
There isn't really anything to be lazy about here... loading the tree loads the full "directory" so the only thing that would be "lazy" would be the selection of the name.